### DISCUSSION TITLE
IMB-1215 CPU Fan not visible

### DISCUSSION THREAD

**Discussion by SamdCastle** [LibreHardwareMonitor/LibreHardwareMonitor#1417] [Q&A]
Hello!
I am very new here so still trying to figure my way around the ways of GitHub, so bare with me please!

I have an Asrock IMB-1215 motherboard and am trying to gain control of the CPU fan. I have exhausted my list of software options, then stumbled across the IMB-1222 support pull request and gave it a try (after a lot of learning of how to make that happen!)

I have managed to get the fan controller to be displayed under the motherboard drop down 'Nuvoton NCT6116D' yet still no luck seeing the CPU_FAN1 speed. The fan speed is shown in BIOS so i'm sure there must be a way to find it, and i'm hoping you guru's might be able to help me please?

Thanks in advance
Sam

@Tellinq - Thanks for your help and patience so far!

![Untitled](https://github.com/user-attachments/assets/2104e2b5-460f-4639-8df9-2555226f9938)

**Comment by Tellinq** [comment]
I just made a branch for the NCT6116D changes and noticed that the fan, voltages, etc. amounts weren't actually defined

Try this build I just set up: https://github.com/Tellinq/LibreHardwareMonitor/actions/runs/10312799857 (any of the `net` zip folders will do, but net472 is usually where I test). Hopefully that should show some sensors now!

If the fans do get detected, they'll still be generic. I'd like to identify which fan is which once we get to that point

**Comment by SamdCastle** [comment]
@Tellinq you beauty!

![image](https://github.com/user-attachments/assets/3f5f6d16-d7fa-48ff-a8dd-b48e03a066b7)

**Reply by Tellinq** [reply]
Excellent! Now if I am to add your board properly, I’d like to know:

- Which fan is the CPU fan? We can assume that the other index is for the Chassis Fan regardless if it’s used. I’m guessing `#2` is the CPU fan on the actual CPU_FAN header, but I’d just like to confirm before assuming anything.
- I’ll remove Fan `#3` for your board specifically as that doesn’t actually exist on your board
- In the BIOS/UEFI, are there any temperatures that show up when messing with the Monitor tab or wherever you manage fan curves if that exists? I’m not entirely sure how ASRock’s industrial UEFI differs in terms of interface if at all compared to modern consumer boards
    - I mainly need to see where the CPU Core and Motherboard/System temperature index aligns. If you’re not sure, using a stress test software could also help give indicators
- I typically ask people to also provide a screenshot of another monitoring software - either an official software utility from ASRock (or the board manufacturer I’m dealing with) or through other popular sensor monitor utilities like HWiNFO64 (typically has the highest likelihood of showing data that may align with your data entry). This is for me to get a good idea of where certain sensors align with the limited amount of interactions I can do without physically being able to use the motherboard in question

**Comment by SamdCastle** [comment]
Correct!
Fan 1 = CHA_FAN1
Fan 2 = CPU_FAN1

I have limited monitoring controls in BIOS. Options are "FULL ON" and "Automatic Mode"

Here is a Log from HWiNFO64 showing Motherboard and Sensors, let me know if that's not showing you what you need to see!

[PS.LOG](https://github.com/user-attachments/files/16569009/PS.LOG)

![240810091016](https://github.com/user-attachments/assets/102f4628-8b0f-477c-9032-154aaf46d1e8)

![hwinfo](https://github.com/user-attachments/assets/f979afc0-e3e1-47e4-9f95-f8ef56d2c671)

**Reply by Tellinq** [reply]
Sweet! I however am a bit skeptical of the temperature sources, namely the CPU temperature.

Fortunately I did manage to decipher all the voltage indexes, except for DC_IN as I'm not completely sure if that will show up.

Try this build I set up: https://github.com/Tellinq/LibreHardwareMonitor/actions/runs/10332208191

- Fan controls should be all properly named now
- Voltages should be properly assigned now
- Temperatures are based off HWiNFO64's names for now

Here's what I need now:
A screenshot of LHM when a CPU stress test is active - showing both all the motherboard details and the i9-9900K temperatures. If multiple temperatures dramatically increase and are similar, keep an eye out for which one responds faster. Whichever temperature source responds faster and is as accurate as what the CPU temperatures directly from your CPU cores report, then that's the temperature source I will want to use in the end.

(Oh also, if you want to take this time to also mess around with FanControl as you initially mentioned in the PR, copy the `LibreHardwareMonitor.dll` and `LibreHardwareMonitor.pdb` files in the dedicated LibreHardwareMonitor folder to FanControl and replace the contents from there)

**Comment by SamdCastle** [comment]
![image](https://github.com/user-attachments/assets/b721a1c5-f7c3-48e2-bf8f-4f9027aa17d2)

Thanks again Tellinq!

So i'm not familiar with stress testing software these days so I just chucked OCCT on, hope that's enough! Temperatures seemed more or less consistent between OCCT and LHM. OCCT showed CPU Core (PECI) which LHM always reads 0. Also motherboard temp on LHM often flashes between 0 and an actual reading. Neither are issues for me of course, I am just giving you a run down of things I noticed.

If this isn't enough, please say. I'm happy to run as many tests as you like! :)

Had a play with Fan Control, exactly what I was hoping for thank you!

**Reply by Tellinq** [reply]
Yup, this should be enough for me! Interesting how the Motherboard and CPU PECI temperatures just stopped working outright. However what I was looking for is to see if my "Temperature `#1`" index would show up and give any meaningful value - looks like that's not the case. Not much I can do about the temperature anomalies, but I'll leave it to the original person who made the PR to look into in case similar behavior is observed.

My work here should be complete now! Thank you for testing, and I'll make sure your board data entry gets merged once either #1376 gets merged or if my changes get merged into that PR

**Reply by Tellinq** [reply]
Actually, just for sanity check - I'm going to remove that index that isn't appearing and see if that makes the temperature anomaly go away. I doubt it but worth a shot.

Should be last build: https://github.com/Tellinq/LibreHardwareMonitor/actions/runs/10335327431

I'm now looking for where the Motherboard temperature aligns. Either the label I put for Motherboard should be that output, or CPU PECI should be. If neither, I'll just keep as is and I'll just merge what I have.

**Comment by SamdCastle** [comment]
Strangely, CPU PECI is is always 0 and Motherboard shows a reading or shows 0, just flicks between the 2 randomly.
Added a screenshot of your latest build, with motherboard temp showing a reading. The motherboard CPU Core temp is quite different to the temps shown under CPU.

Also, not sure what the Intel PCH reading is at the bottom, but it idles at around 75c then drops to about 65c during a stress test.

![4etd](https://github.com/user-attachments/assets/3497a0cc-9384-4526-aaef-e8a1ec814bd4)

**Reply by Tellinq** [reply]
This stumps me, however for the time being I think I have enough data to have a proper data entry for your board. The rest of these issues a bit out of my league for now. For the time being, I’ll get the data entry for your board PR’d so at the very least you’ll have useful fan data/control and voltage sensors.

The temperature readings will likely have to be tackled at a later time, but as long as the CPU readouts from your i9-9900K are reliable, I still consider this implementation a success, while rather rocky.

**Reply by SamdCastle** [reply]
Absolutely, you have solved my issue and it is plenty good enough for my purposes.

Thank you once again for all your help, absolute legend!
